lib/lz4frame_static.h

```diff
@@ -82,6 +82,60 @@ typedef enum { LZ4F_LIST_ERRORS(LZ4F_GENERATE_ENUM) } LZ4F_errorCodes;
 LZ4F_errorCodes LZ4F_getErrorCode(size_t functionResult);
 
 
+
+/**********************************
+ *  Bulk processing dictionary API
+ *********************************/
+typedef struct LZ4F_CDict_s LZ4F_CDict;
+
+/*! LZ4_createCDict() :
+ *  When compressing multiple messages / blocks with the same dictionary, it's recommended to load it just once.
+ *  LZ4_createCDict() will create a digested dictionary, ready to start future compression operations without startup delay.
+ *  LZ4_CDict can be created once and shared by multiple threads concurrently, since its usage is read-only.
+ * `dictBuffer` can be released after LZ4_CDict creation, since its content is copied within CDict */
+LZ4F_CDict* LZ4F_createCDict(const void* dictBuffer, size_t dictSize);
+void        LZ4F_freeCDict(LZ4F_CDict* CDict);
```

Comment:
Good point.
The `LZ4F_errorCode_t` was there in case a `free` operation could fail.
But effectively, it never fails, so the return type is not very useful.
In fact, it creates additional charge into the user code, because now it adds to add some test and special exception code in case the function would fail.

As a consequence, nowadays, I'm more in the mood of having `free` function with a `void` type, to make it clear it never fails, and does not need extra-care.

That being said, it's true that this choice makes this prototype stand apart from other `free` functions defined in `lz4frame`. And it would be a good reason to use the same return type, even if the function always succeeds.

Which choice is better ? Well, I'm not sure. Feedback welcomed.